server/src/main/java/com/sauron/account/User.java

```diff
@@ -27,19 +26,16 @@
     @Column(unique = true, nullable = false)
     private String email;
 
-    @ElementCollection
-    @CollectionTable(
-            name = "BANK_CONNECTION_DATA",
-            joinColumns = @JoinColumn(name = "USER_ID")
-    )
-    private Set<BankConnectionData> bankConnectionData = new HashSet<>();
+    @OneToMany
+    @JoinColumn(name = "USER_ID")
+    private Set<BankAccount> bankAccounts = new HashSet<>();
```

Comment:
User and BankAccount have two-way association (and BankAccount relation is implicit - there is no proxy for User element). This can be quite problematic because developer should be responsible for model consistency. For example when deleting user, developer should also look for every account and delete it manually. Also creation new account means that developer has to manually add it to user.. Maybe this can be change somewho?

Reply:
I would also add here @Cascade annotation with at least PERSIST option